crates/biome_js_analyze/src/react/hooks.rs

```diff
@@ -474,6 +474,30 @@ pub fn is_binding_react_stable(
     })
 }
 
+/// Unwrap the expression to a call expression without changing the result of the expression,
+/// such as type assertions.
+fn unwrap_to_call_expression(expression: AnyJsExpression) -> Option<JsCallExpression> {
+    match expression {
+        AnyJsExpression::JsCallExpression(expr) => Some(expr),
+        AnyJsExpression::JsParenthesizedExpression(expr) => {
+            expr.expression().ok().and_then(unwrap_to_call_expression)
```

Comment:
This is a recursion, which we tend to avoid when working with CSTs. Can you make it a loop? It shouldn't be too difficult.

Reply:
@ematipico Sorry, I didn't know that. I turned it into a loop in [45e9668](https://github.com/biomejs/biome/pull/4964/commits/45e96689950cd85bc863dcae9a8c039cee41a3b3).